packages/haul-babel-preset-react-native/src/index.ts

```diff
@@ -1,49 +1,137 @@
 import path from 'path';
 
-const defaultPlugins = [
-  [require('@babel/plugin-proposal-class-properties'), { loose: true }],
-  [require('@babel/plugin-proposal-optional-catch-binding')],
-  [require('@babel/plugin-syntax-dynamic-import')],
-  [require('@babel/plugin-syntax-export-default-from')],
-  [require('@babel/plugin-transform-react-jsx')],
-  [require('@babel/plugin-transform-sticky-regex')],
-  [require('@babel/plugin-transform-unicode-regex')],
-  // For some reason native async/await don't behave correctly
-  // on RN 0.59 on both platforms, so we need to transpile it
-  // to native Promises.
-  [require('./transforms/superMemberArrowFunction').default],
-  [require('@babel/plugin-transform-async-to-generator')],
-];
-
-// Additional plugins for Hermes because it doesn't support ES6 yet
-const hermesPlugins = [
-  [require('@babel/plugin-transform-classes')],
-  [require('@babel/plugin-transform-shorthand-properties')],
-  [require('@babel/plugin-transform-template-literals'), { loose: true }],
-];
-
-const chakraPlugins = [
-  [require('@babel/plugin-transform-spread')],
-  [require('@babel/plugin-proposal-object-rest-spread')],
-];
-
-const commonJsPlugin = [
-  require('@babel/plugin-transform-modules-commonjs'),
-  { allowTopLevelThis: true },
-];
-
-function isTypeScriptSource(fileName: string) {
+export function isTypeScriptSource(fileName: string) {
   return !!fileName && fileName.endsWith('.ts');
 }
 
-function isTSXSource(fileName: string) {
+export function isTSXSource(fileName: string) {
   return !!fileName && fileName.endsWith('.tsx');
 }
 
-function isReactNative(fileName: string) {
+export function isReactNative(fileName: string) {
   return !!fileName && fileName.includes(`node_modules${path.sep}react-native`);
 }
 
+type PluginSpec = {
+  name: string;
+  options?: object;
+};
+
+const commonJsPlugin: PluginSpec = {
+  name: '@babel/plugin-transform-modules-commonjs',
+  options: { allowTopLevelThis: true },
+};
+
+export function getDefaultPrePlugins(): PluginSpec[] {
+  return [
+    {
+      // The flow strip types plugin must go BEFORE class properties!
+      name: '@babel/plugin-transform-flow-strip-types',
+    },
+    {
+      name: '@babel/plugin-proposal-class-properties',
+      options: { loose: true },
```

Comment:
Wouldn't it be easier to conform to Babel configuration scheme instead of creating new and mapping it back to Babel? So PluginSpec would be something like `string | [string, object]`.

Reply:
I think it's easier to use object or `[string, object]` tuple for consuming `@haul-bundler/babel-preset-react-native` as a base for custom preset.

Reply:
I'm fine as long as it's documented. Just want you to keep an eye on limiting the API surface, it's a pain to learn different config schemes for different tools and we should avoid it.